bundler/lib/bundler/settings.rb

```diff
@@ -595,5 +596,18 @@ def self.key_to_s(key)
         end
       end
     end
+
+    def credentials_from_helper(uri)
+      helper_key = "credential-helper.#{uri.host}"
+      helper_path = self[helper_key]
+      return unless helper_path
+
+      begin
+        output = `#{helper_path}`.strip
```

Comment:
@segiddins do you see any security issues with calling out this? Value is coming from bundle config.

Reply:
I have a preference that we use the same pattern as https://git-scm.com/docs/gitcredentials#_configuration_options. Additionally, we should be using `IO.popen` for the subprocess

Reply:
@simi @segiddins 0e48def...be95a01

Taking security concerns into consideration, I have modified the code to use IO.popen.

Additionally, similar to Git, I have implemented function a that automatically recognizes executables named bundler-credential-{helper-name} for automatically discovering helpers.